nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -57,6 +57,7 @@ def __init__(
             self.add_special_tokens(special_tokens)
 
     def text_to_tokens(self, text):
+        text = text.rstrip("\n")
```

Comment:
This affects other domains such as ASR. Needs to be backward compatible for it.

Reply:
Do these models make use of a trailing `\n`? As described yttm and sentencepiece would remove it by default.

Reply:
Depends on the dataset. And it's varied so by default it should be old behavior.

Reply:
Got it, thanks!

Reply:
Changed the code such that trailing newlines will only be stripped for machine translation datasets.